Profile epic

roma685235423 opened the pull request.

alrzi committed:
prepared common part of the project

roma685235423 committed:
[Профиль] Сверстать экран "Профиль"

Создал ProfileViewController, ячейку ProfileTableViewCell для таблицы навигации информации профиля.

miamib34ch committed:
Добавил ассет кнопки сортировки

miamib34ch committed:
Удалил лишний ассет, поскольку цвет можно менять в коде

miamib34ch committed:
Merge pull request #55 from miamib34ch/feature_sortButtonImage

Добавил ассет кнопки сортировки

roma685235423 committed:
[Профиль] Сверстать экран "Профиль"

Создал ProfileNetworkClient и ProfileViewPresenter. Реализовал с их помощью загрузку данных профиля и фото. В ProfileViewController добавил activityIndicator и логику его отображения во время загрузки фото и скрытия после в ProfileViewPresenter.

roma685235423 committed:
[Профиль] Сверстать экран "Профиль"

Реализовал загрузку и отображение всех данных пользователя и изменение заголовков ячеек в зависимости от ответа

roma685235423 committed:
[Профиль] Сверстать экран "Редактирование профиля"

Сверстал экран редактирования профиля, создал кастомные поля редактирования данных, добавил алерт с полем ввода ссылки и алерт неуспешной проверки ссылки, так же произвёл небольшие изменения в именах сущностей.

roma685235423 committed:
[Профиль] Сверстать экран "Редактирование профиля"

Реализовал передачу редактированных данных профиля в экран профиля для их отправки с помощью сетевого слоя презентера

roma685235423 committed:
[Профиль] Обновление данных на сервере.

Реализована логика отправки обновлённых данных на сервер после редактирования на экране. С помощью ProcessHUD добавлен индикатор загрузки, и блокировка интерфейса.

miamib34ch committed:
Добавил сервис для блокировки UI

roma685235423 committed:
Merge pull request #57 from miamib34ch/feature_UIBlockingProgressHUD

Добавил сервис для блокировки UI

roma685235423 committed:
Merge pull request #58 from roma685235423/profile_epic

Profile epic

roma685235423 committed:
Merge branch 'base' into profile_epic

roma685235423 commented:
Ссылка на таск-трекер

https://github.com/users/miamib34ch/projects/1

miamib34ch commented on FakeNFT/Epics/Profile/UIBlockingProgressHUD.swift:
```diff
@@ -0,0 +1,20 @@
+import UIKit
```
Повторяется код

miamib34ch reviewed (commented):
Хорошая работа, всё круто, только код сервиса progresshud повторяется

roma685235423 committed:
[Профиль] Создать веб-вью для просмотра сайта разработчика

Добавил UserAboutWebView и реализовал модальное открытие веб вью при нажатии на лэйбл сайта или ячейку "О разработчике"

alrzi commented:
для регистрации ячеек и их создание можно использовать экстеншены из CellsReusingUtils

roma685235423 committed:
[Профиль] Сверстать экран "Мои NFT"

Приступил к вёрстке экрана "Мои NFT"

roma685235423 commented:
Done

roma685235423 committed:
[Профиль] Сверстать ячейку для UITableView экрана "Мои NFT"

Приступил к созданию ячейки. Создал модели ответа для сущности автор, NFT, а так же для ячейки. Так же в MyNFTsViewController добавил TableView.

roma685235423 committed:
[Профиль] Реализовать логику загрузки данных UI элементов экрана "Мои NFT".

Преступил к созданию сервиса загрузки данных для экрана и презентера.

roma685235423 committed:
[Профиль] Сверстать ячейку для UICollectionView экрана "Избранные NFT"

Сверстал ячейку, UICollectionView. В процессе ViewController и Presenter

roma685235423 committed:
[Профиль] Реализовать загрузку данных "Мои NFT".

Реализовал загрузку данных и их обработку для слоя представления экрана "Мои NFT".
